### PR TITLE
add property view and upsert

### PR DESCRIPTION
**Pull request by miketynes**
~need to resolve issue w/ actor: the actor_uuid from parameter_def seems to override the one in parameter. The trouble is I patterned my code after property_def/property, so this issue may be there as well. We should take a look~

Resolved: it was a small mistake in my join.

### CODE REVIEW

File: data_model/sql_core/prod_views.sql [hunk 1]
```diff
@@ -987,12 +987,28 @@ EXECUTE PROCEDURE upsert_parameter_def ( );
...
---
+SELECT
+	pr.parameter_uuid,
+	pr.parameter_def_uuid,
+    pd.description as parameter_def_description,
+	pr.parameter_val,
+	pr.actor_uuid,
+    act.description as actor_description,
+	pr.status_uuid,
+    st.description as statsus_description,
```
Comment (gcatabr1) [inline data_model/sql_core/prod_views.sql:998]: status

File: data_model/sql_core/prod_views.sql [hunk 2]
```diff
@@ -987,12 +987,28 @@ EXECUTE PROCEDURE upsert_parameter_def ( );
...
+	pr.add_date,
+	pr.mod_date
```
Comment (gcatabr1) [inline data_model/sql_core/prod_views.sql:1000]: need a way to link this record to an entity

File: data_model/sql_core/prod_views.sql [hunk 3]
```diff
@@ -987,12 +987,28 @@ EXECUTE PROCEDURE upsert_parameter_def ( );
...
 FROM parameter pr
-;
-
+LEFT JOIN parameter_def pd on pr.parameter_def_uuid = pd.parameter_def_uuid
+LEFT JOIN actor act on pd.actor_uuid = act.actor_uuid
+LEFT JOIN status st on pd.status_uuid = st.status_uuid;
```
Comment (gcatabr1) [inline data_model/sql_core/prod_views.sql:1004]: join the property_x table

File: data_model/sql_core/prod_upserts.sql [hunk 4]
```diff
@@ -1470,34 +1470,71 @@ Name:			upsert_parameter()
...
+ 				delete from vw_parameter where parameter_def_description = 'beard_moisturize_dur';
  */
 CREATE OR REPLACE FUNCTION upsert_parameter()
 	RETURNS TRIGGER
 	AS $$
 BEGIN
 	IF(TG_OP = 'DELETE') THEN
-
+	    -- first delete the parameter record
+		DELETE FROM parameter
```
Comment (gcatabr1) [inline data_model/sql_core/prod_upserts.sql:1500]: will need to delete the 'joining' record first...

**Comment by gcatabr1** [comment]
take a look at comments regarding the 'joining' table property_x. this will need to be added to the view as well as addressed in the delete (and insert).

File: data_model/sql_core/prod_upserts.sql [hunk 5]
```diff
@@ -1470,34 +1470,71 @@ Name:			upsert_parameter()
...
+        IF (select exists
+                (select parameter_def_uuid
+                 from vw_parameter_def
+                 where parameter_def_uuid = NEW.parameter_def_uuid)
+            )
+        THEN
+			INSERT INTO parameter (parameter_def_uuid, parameter_val, actor_uuid, status_uuid)
+				VALUES(NEW.parameter_def_uuid, NEW.parameter_val, NEW.actor_uuid, NEW.status_uuid)
+				returning parameter_uuid into NEW.parameter_uuid;
+			RETURN NEW;
```
Comment (gcatabr1) [inline data_model/sql_core/prod_upserts.sql:1529]: will need to address the insert into the 'joining' table property_x

**Comment by miketynes** [comment]
Merged following in-person review and changes w/ @gcatabr1

**Pull request merged**